### PR TITLE
Gut a hidden GUC optimizer_enable_partial_index

### PR DESCRIPTION
**Pull request by d**
The feature is undergoing major rethink, and is in our way of something
more important.

While we were at it, also rid the test of `optimizer_plan_id`, it's a
broken way of testing a feature.

### CODE REVIEW

File: src/test/regress/expected/gp_optimizer.out [hunk 1]
```diff
@@ -8228,29 +8228,34 @@ insert into orca.t_date values('01-03-2012'::date,6,'tag1','tag2');
...
-   ->  Sequence  (cost=0.00..437.00 rows=1 width=28)
-         ->  Partition Selector for t_date (dynamic scan id: 1)  (cost=10.00..100.00 rows=50 width=4)
-               Partitions selected:  6 (out of 6)
-         ->  Result  (cost=0.00..437.00 rows=1 width=28)
-               ->  Append  (cost=0.00..437.00 rows=1 width=28)
-                     ->  Dynamic Index Scan on t_date (dynamic scan id: 1)  (cost=0.00..6.00 rows=1 width=58)
-                           Index Cond: user_id = 9::numeric
-                     ->  Dynamic Table Scan on t_date (dynamic scan id: 1)  (cost=0.00..431.00 rows=1 width=58)
-                           Filter: user_id = 9::numeric
+INFO:  GPORCA failed to produce a plan, falling back to planner
```
Comment (vraghavan78) [inline src/test/regress/expected/gp_optimizer.out]: can we know why it is falling back? It should generate a plan nevertheless.

File: src/test/regress/expected/gp_optimizer.out [hunk 2]
```diff
@@ -8228,29 +8228,34 @@ insert into orca.t_date values('01-03-2012'::date,6,'tag1','tag2');
 insert into orca.t_date values('01-03-2012'::date,7,'tag1','tag2');
 insert into orca.t_date values('01-03-2012'::date,8,'tag1','tag2');
 insert into orca.t_date values('01-03-2012'::date,9,'tag1','tag2');
-set optimizer_enable_partial_index=on;
 set optimizer_enable_space_pruning=off;
 set optimizer_enable_constant_expression_evaluation=on;
 set optimizer_enumerate_plans=on;
 set optimizer_plan_id = 2;
```
Comment (vraghavan78) [inline src/test/regress/expected/gp_optimizer.out]: this is why. Can we remove this plan space id? This is a broken way of testing a feature.

File: src/test/regress/expected/gp_optimizer.out [hunk 3]
```diff
@@ -8289,23 +8294,23 @@ set optimizer_enable_constant_expression_evaluation=on;
...
```
Comment (vraghavan78) [inline src/test/regress/expected/gp_optimizer.out]: remove this.

**Comment by d** [comment]
Jemish and I just updated the test to address the comment about `optimizer_plan_id`

**Pull request merged**